### PR TITLE
support default branch detection for non-GitHub repositories

### LINKED ISSUE #4937: Branch delete only works if there is a local master branch
**Version:** 1.2.3
**OS:** Windows

In my GitLab installation I have a repository with the branch "Develop" marked as default.

If I now use Github Desktop and make a clone, I get the develop branch. That's correct, too.

If I now create a branch feature/xy on develop and push it I can't delete it anymore. The menu item under Branch is deactivated.

Only when I have origin/master locally can I delete the branch I created.

As soon as I have a master branch locally in the dialog "Create Branch" the master is shown as default, which is wrong.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #4937

I wanted this to be a careful change so I focused on this flow:

 - use`git symbolic-ref refs/remotes/{remote}/HEAD` and ensuring that points to a valid ref
 - strip the returned `refs/remotes/{remote}/{branch}` to just `{branch}` to match the conventions for GitHub default branch detection
 - fallback to `master` if this doesn't work

With this change, the default branch now lights up in the list:

And the menu items are now enabled for non-default branches:

And the default branch still inherits the "cannot rename, cannot delete" convention in the app:

FYI: I'm using [this repo](https://gitlab.com/gitlab-org/gitter/webapp) for testing, for whoever ends up reviewing this.

### CODE REVIEW

File: app/src/lib/stores/git-store.ts [hunk 1]
```diff
@@ -358,11 +359,33 @@ export class GitStore extends BaseStore {
...
+
+      if (match != null && match.startsWith(remoteNamespace)) {
+        // strip out everything related to the remote because this
+        // is likely to be a tracked branch locally
+        // e.g. `master`, `develop`, etc
+        defaultBranchName = match.substr(remoteNamespace.length)
+      }
+    }
+
+    if (defaultBranchName == null) {
```
[the remark this answers is not in the document]
Comment (shiftkey) [reply app/src/lib/stores/git-store.ts]: This check becomes unnecessary now that `resolveDefaultBranch()` always returns a string :fire:

**Comment by shiftkey** [comment]
🌔

File: app/src/lib/git/refs.ts [hunk 2]
```diff
@@ -23,6 +23,19 @@ export function formatAsLocalRef(name: string): string {
...
+ * Read a symbolic ref from the repository.
+ *
+ * Symbolic refs are used to point to other refs, similar to how symlinks work
+ * for files. Because refs can be removed easily from a Git repository,
+ * symbolic refs should only be used when absolutely necessary.
+ *
+ * @param repository The repository to lookup
+ * @param ref The symbolic ref to resolve
+ *
+ * @returns the resolved ref, if found, or `null` if `ref` cannot be found or
```
Comment (niik) [inline app/src/lib/git/refs.ts]: Nitpick but in Git terminology a 'resolved' ref usually means a SHA. I think we should reword this to make sure there's no ambiguity around what this returns, i.e. it will return a canonical ref.

**Comment by shiftkey** [comment]
:apple:
